lib/client/protocols/arrays/sharing.js

```diff
@@ -1,123 +1,169 @@
-   * This function will reveal the lengths of the shared array
-   *
-   * If parties would like to keep the lengths of their arrays secret, they should agree on some "max" length apriori (either under MPC
-   * or as part of the logistics of the computation), all their arrays should be padded to that length by using appropriate default/identity
-   * values
-   * @function jiff_share_array
-   * @ignore
-   * @param {module:jiff-client~JIFFClient} jiff - the jiff client instance.
-   * @param {Array} array - the array to be shared.
-   * @param {null|number|object} lengths - the lengths of the arrays to be shared, has the following options:
-   *                                       1. null: lengths are unknown, each sender will publicly reveal the lengths of its own array.
-   *                                       2. number: all arrays are of this length
-   *                                       3. object: { <sender_party_id>: length }: must specify the length of the array for each sender.
-   * @param {number} [threshold=receivers_list.length] - the min number of parties needed to reconstruct the secret, defaults to all the receivers.
-   * @param {Array} [receivers_list=all_parties] - array of party ids to share with, by default, this includes all parties.
-   * @param {Array} [senders_list=all_parties] - array of party ids to receive from, by default, this includes all parties.
-   * @param {number} [Zp=jiff.Zp] - the mod.
-   * @param {string|number} [share_id=auto_gen()] - the base tag used to tag the messages sent by this share operation, every element of the array
-   *                                   will get a unique id based on the concatenation of base_share_id and the index of the element.
-   *                                   This tag is used so that parties distinguish messages belonging to this share operation from
-   *                                   other share operations between the same parties (when the order of execution is not
-   *                                   deterministic). An automatic id is generated by increasing a local counter, default
-   *                                   ids suffice when all parties execute all sharing operations with the same senders
-   *                                   and receivers in the same order.
-   * @return {promise} if the calling party is a receiver then a promise to the shared arrays is returned, the promise will provide an object
-   *                    formatted as follows: { <party_id>: [ <1st_share>, <2nd_share>, ..., <(lengths[party_id])th_share> ] }
-   *                    where the party_ids are those of the senders.
-   *                    if the calling party is not a receiver, then null is returned.
-   */
-  jiff_share_array: function (jiff, array, lengths, threshold, receivers_list, senders_list, Zp, share_id) {
-    var i;
-
+const share_array = function (jiff, array, lengths, threshold, receivers_list, senders_list, Zp, share_id) {
```

Comment:
This file looks a lot lot better now. But let us do two more simple things:
1. make sure there is an empty line between every function
2. this file is still quite big, let us split it into a few files: for example:
  a. api.js just requires all other files and exposes the right functions, this is what the lib/client/api/sharing.js requires.
  b. share.js includes the main share_ND_array function, the share_array and share_2D_array wrappers, share_ND_array_deferred , share_ND_array_static, share_array_single_sender 
c. open.js includes the open_array and open_ND_array functions and the receive_open_ND_array etc (also maybe add a useless open_2D_array wrapper for consistency).
d. util.js includes the remaining couple of functions (the ones about sanitizing parameters and matching skeletons etc)

Reply:
lib/client/protocols/arrays/api.js
```javascript
const share = require('./share.js');
const open = require('./open.js');
const util = require('./util.js');

module.exports = {
  jiff_share_array: share.share_array,
  jiff_share_2D_array: share.share_2D_array,
  jiff_share_ND_array: share.share_ND_array,
  jiff_share_ND_array_static: share.share_ND_array_static,
  jiff_share_ND_array_deferred: share.share_ND_array_deferred,

  jiff_skeleton_of: util.skeleton_of,

  jiff_open_array: open.open_array,
  jiff_open_ND_array: open.open_ND_array,
  jiff_receive_open_ND_array: open.receive_open_ND_array
};
```

Reply:
That's great!!